### PR TITLE
Companion for paritytech/polkadot#6117

### PR DESCRIPTION
**Pull request by altonen**

### CODE REVIEW

**Review by dmitry-markin — APPROVED** [review]

File: client/relay-chain-minimal-node/src/network.rs
Comment (skunert) [inline client/relay-chain-minimal-node/src/network.rs:137]: This does not need to be on the `DummyChainSync` no?

Comment (altonen) [reply client/relay-chain-minimal-node/src/network.rs:137]: It doesn't have to be but it's part of `ChainSync` in Substrate so I thought making it part of `DummyChainSync` would closest to what we have in Substrate

**Comment by altonen** [comment]
> Since the substrate PR already merged you probably need to update substrate to make CI happy

Could you elaborate? This is a companion for a [polkadot companion](https://github.com/paritytech/polkadot/pull/6117) (which is showing green for this) so unless I've misunderstood something about the process, CI should be happy once the Polkadot PR is merged and this PR's references are updated accordingly.

**Comment by skunert** [comment]
> > Since the substrate PR already merged you probably need to update substrate to make CI happy
> 
> Could you elaborate? This is a companion for a [polkadot companion](https://github.com/paritytech/polkadot/pull/6117) (which is showing green for this) so unless I've misunderstood something about the process, CI should be happy once the Polkadot PR is merged and this PR's references are updated accordingly.

Ah yes, should be fine then :+1:

File: client/relay-chain-minimal-node/src/network.rs
Comment (skunert) [reply client/relay-chain-minimal-node/src/network.rs:137]: On second thought, it might make more sense to have a separate method in substrate that exposes this. In case something changes we want the cumulus implementation to follow.

Comment (altonen) [reply client/relay-chain-minimal-node/src/network.rs:137]: I believe the vision is to make it more transparent so that you could instantiate some syncing engine implementation that would then install the necessary notification and request protocols to, e.g., `sc_network::Params` so that there would be no need for calling, e.g. `light_client_requests::generate_protocol_config()` (except for in the syncing implementation of course) but I'm not opposed to exposing this through Substrate as a temporary measure.

The fire I'm trying to put out right now though is that Polkadot CI is blocked by this PR. How do you want to proceed with this?

Comment (skunert) [reply client/relay-chain-minimal-node/src/network.rs:137]: Okay, I think its fine for now then, we could have a follow up.

**Review by skunert — APPROVED** [review]

**Pull request merged**
